validate/validate.go

```diff
@@ -206,11 +206,6 @@ func (v *Validator) CheckRoot() (errs error) {
 		return
 	}
 
-	if filepath.Base(v.spec.Root.Path) != "rootfs" {
-		errs = multierror.Append(errs,
-			specerror.NewError(specerror.PathName, fmt.Errorf("path name should be the conventional 'rootfs'"), rspec.Version))
```

Comment:
This is a [spec SHOULD][1], and removing it is why [your Travis tests fail][2].  I think we need to keep this check, and users that don't care about SHOULD-level violations can ignore those errors [like this][3].

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.0/config.md#L41
[2]: https://travis-ci.org/opencontainers/runtime-tools/jobs/282233019#L502
[3]: https://github.com/opencontainers/runtime-tools/blob/v0.2.0/cmd/runtimetest/main.go#L767

Reply:
I know it's a SHOULD. But `CheckAll` doesn't appear to understand that SHOULD doesn't mean MUST. The following error happens:

```
=== RUN   TestGenerateValid
--- FAIL: TestGenerateValid (0.57s)
        generate_test.go:42: unexpected error(s) validating default: 3 errors occurred:

                * path name should be the conventional 'rootfs'
                Refer to: https://github.com/opencontainers/runtime-spec/blob/v1.0.0/config.md#root
                * root.path is "", but it MUST be a child of "/tmp/TestGenerateValid_bundle231889172"
                Refer to: https://github.com/opencontainers/runtime-spec/blob/v1.0.0/bundle.md#container-format
                * 'Root.Path' should not be empty
```

And this means that `oci-runtime-tool validate` returns a non-zero error code.

Reply:
> But `CheckAll` doesn't appear to understand that SHOULD doesn't mean MUST.

`CheckAll` shouldn't care; it's up to the caller to decide what is fatal or not.  Using `rootfs` in the default config should be enough for your `TestGenerateValid`.  `cmd/oci-runtime-tool/validate.go` needs a `--compliance-level` like `runtimetest` (which may be in flight somewhere, I don't have my dev box handy to grep open PRs).

Reply:
> `cmd/oci-runtime-tool/validate.go` needs a `--compliance-level` like `runtimetest` (which may be in flight somewhere, I don't have my dev box handy to grep open PRs).

I couldn't find this in flight, so I filed it as #492.